### PR TITLE
feat: make blockchain scan interval configurable

### PR DESCRIPTION
**Pull request by JoeCap08055**
- add BLOCKCHAIN_SCAN_INTERVAL_SECONDS config
- add TRUST_UNFINALIZED_BLOCKS config
- add unit tests
- refactor BlockchainScannerService to use a throw pattern instead of boolean functions for checking scan validity

### CODE REVIEW

File: env.template [hunk 1]
```diff
@@ -24,6 +24,13 @@ PROVIDER_BASE_URL=http://host.docker.internal:5555/api/v3/webhook
 REDIS_URL=redis://0.0.0.0:6379
 #REDIS_URL=redis://redis:6379                        #docker dev
 
+# How many seconds to delay between successive scans of the chain
```
Comment (mattheworris) [inline env.template:27]: We should add these new env vars to `docker-compose.yaml`

**Review by mattheworris — APPROVED** [review]
- Read through changes
- Executed Unit Tests (All passed)

🚢 it!

File: libs/common/src/blockchain/blockchain.service.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export class BlockchainService implements OnApplicationBootstrap, OnApplicationS
     return this.api.rpc.chain.getBlockHash(block);
   }
 
-  public getBlock(block: BlockHash): Promise<SignedBlock> {
+  public getBlock(block?: BlockHash): Promise<SignedBlock> {
```
Comment (claireolmstead) [inline libs/common/src/blockchain/blockchain.service.ts:98]: Nit: I feel like the check for if the block exists or not should be before we call this function. Seems like every time we call getBlock, we can assume that we have an existing blockhash.

**Review by claireolmstead — APPROVED** [review]
LGTM

File: env.template (hunk 1 above)
Comment (JoeCap08055) [reply env.template:27]: Done

File: libs/common/src/blockchain/blockchain.service.ts (hunk 2 above)
Comment (JoeCap08055) [reply libs/common/src/blockchain/blockchain.service.ts:98]: That's not the point; if you call with no arguments you get the latest block

**Pull request merged**